torchrl/envs/transforms/transforms.py

```diff
@@ -4568,15 +4593,20 @@ def transform_observation_spec(
             )
         for key, spec in self.primers.items():
             if spec.shape[: len(observation_spec.shape)] != observation_spec.shape:
-                raise RuntimeError(
-                    f"The leading shape of the primer specs ({self.__class__}) should match the one of the parent env. "
-                    f"Got observation_spec.shape={observation_spec.shape} but the '{key}' entry's shape is {spec.shape}."
-                )
+                try:
+                    expanded_spec = self._try_expand_shape(spec)
+                except AttributeError:
+                    raise RuntimeError(
```

Comment:
When will this be reached?

Reply:
if for any reason parent is None

Reply:
when would `transform_observation_spec` be called when `parent` is `None`?